subworkflows/local/bam_variant_calling_sentieon_haplotyper/main.nf

```diff
@@ -0,0 +1,154 @@
+
+    if (joint_germline) {
+        genotype_intervals = SENTIEON_HAPLOTYPER.out.gvcf
+            .join(SENTIEON_HAPLOTYPER.out.gvcf_tbi, failOnMismatch: true)
+            .join(cram_intervals_for_sentieon, failOnMismatch: true)
+            .map{ meta, gvcf, tbi, cram, crai, intervals -> [ meta, gvcf, tbi, intervals ] }
+    }
+
+    // Figure out if using intervals or no_intervals
+    SENTIEON_HAPLOTYPER.out.vcf.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
+        }.set{haplotyper_vcf_branch}
+
+
+    SENTIEON_HAPLOTYPER.out.vcf_tbi.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
+        }.set{haplotyper_vcf_tbi_branch}
+
+    SENTIEON_HAPLOTYPER.out.gvcf.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
+        }.set{haplotyper_gvcf_branch}
+
+    SENTIEON_HAPLOTYPER.out.gvcf_tbi.branch{
+            intervals:    it[0].num_intervals > 1
+            no_intervals: it[0].num_intervals <= 1
+        }.set{haplotyper_gvcf_tbi_branch}
+
+    // VCFs
+    // Only when using intervals
+    MERGE_SENTIEON_HAPLOTYPER_VCFS(
+        haplotyper_vcf_branch.intervals.map{ meta, vcf ->
+            new_meta = [
+                            id:             meta.sample,
+                            num_intervals:  meta.num_intervals,
+                            patient:        meta.patient,
+                            sample:         meta.sample,
+                            sex:            meta.sex,
+                            status:         meta.status,
+                            variantcaller:  "sentieon_haplotyper"
+                        ]
+            [groupKey(new_meta, new_meta.num_intervals), vcf] }
+            .groupTuple(),
+        dict.map{it -> [ [ id:'dict' ], it ]})
+
+    versions = versions.mix(MERGE_SENTIEON_HAPLOTYPER_VCFS.out.versions)
+
+    haplotyper_vcf = Channel.empty().mix(
+        MERGE_SENTIEON_HAPLOTYPER_VCFS.out.vcf,
+        haplotyper_vcf_branch.no_intervals)
+
+    haplotyper_tbi = Channel.empty().mix(
+        MERGE_SENTIEON_HAPLOTYPER_VCFS.out.tbi,
+        haplotyper_vcf_tbi_branch.no_intervals)
+
+    if (!skip_haplotyper_filter) {
+        VCF_VARIANT_FILTERING_GATK(haplotyper_vcf.join(haplotyper_tbi),
+                    fasta,
+                    fasta_fai,
+                    dict,
+                    intervals_bed_combined,
+                    known_sites_indels.concat(known_sites_snps).flatten().unique().collect(),
+                    known_sites_indels_tbi.concat(known_sites_snps_tbi).flatten().unique().collect())
+
+        vcf = VCF_VARIANT_FILTERING_GATK.out.filtered_vcf.map{ meta, vcf-> [[patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals, variantcaller:"sentieon_haplotyper"], vcf]}
```

Comment:
Can't we have a better meta map re-organisation here?

Reply:
you mean indentation and linebreaks? (Just for the record, I'm sure I copied that from somewhere else in the pipeline, I would never write such an insane long line.)

Reply:
No, I mean better than:
```
.map{ meta, vcf-> [[patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals, variantcaller:"sentieon_haplotyper"], vcf]}
```
something like:
```
.map {meta, vcf -> [ meta + variantcaller:"sentieon_haplotyper"], vcf]}
```
should be enough, no?

Reply:
Yes, I think that should do it. Let me just update that.